Template10 (Library)/Services/NavigationService/NavigationService.cs

```diff
@@ -290,8 +299,16 @@ public void Navigate<T>(T key, object parameter = null, NavigationTransitionInfo
             state.Write<string>("NavigateState", FrameFacadeInternal?.NavigationService.NavigationState);
         }
 
-        public event TypedEventHandler<Type> AfterRestoreSavedNavigation;
-        public async Task<bool> RestoreSavedNavigationAsync()
+        public event TypedEventHandler<Type> OnNavigationStateLoaded;
```

Comment:
I also decided to make Save and Load asynchronous. They're not using the new pattern, though.